### PR TITLE
Remove access manager

### PR DESCRIPTION
**Pull request by datosh**
Signed-off-by: Fabian Kammel <fk@edgeless.systems>

### Proposed change(s)

- Access manager was removed from deployment some time ago. This change will also remove all traces of access manager and SSH user+key management from code base.
- A Kubernetes native way on accessing nodes was documented. 

### Checklist

- [ ] Update [CHANGELOG.md](https://github.com/edgelesssys/constellation/blob/main/CHANGELOG.md)
- [ ] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [ ] Link to Milestone

### CODE REVIEW

**Review by malt3 — APPROVED** [review]
That is a very thorough cleanup and a nice reduction of our codebase. LGTM

File: docs/docs/workflows/troubleshooting.md
Comment (malt3) [inline docs/docs/workflows/troubleshooting.md:56]: I like the use of ephemeral containers here.

**Review by 3u13r — APPROVED** [review]
LGTM, please sync with @daniel-weisse before merging.

**Comment by malt3** [comment]
Just FYI: removing the ssh field from the config struct is technically a breaking change since you cannot take your old config and reuse it. We may want to either keep it in the config but return a warning/error that is very descriptive OR remove it from the config but tell the user exactly what to do if parsing fails.

**Comment by datosh** [comment]
> Just FYI: removing the ssh field from the config struct is technically a breaking change since you cannot take your old config and reuse it. We may want to either keep it in the config but return a warning/error that is very descriptive OR remove it from the config but tell the user exactly what to do if parsing fails.

Ohh good point! I didn't realize that additional fields will also make the parsing fail. I will get this fixed!

**Comment by daniel-weisse** [comment]
Changes look good to me.
As this changes or removes a lot of code that is also changed or removed by https://github.com/edgelesssys/constellation/pull/387 and https://github.com/edgelesssys/constellation/pull/477 I would like to wait at least until the first of the PRs is merged before bringing this PR to main.

**Comment by datosh** [comment]
Sounds good! I'll wait until #387 is merged.

**Comment by daniel-weisse** [comment]
#387 is merged, you will have to fix merge conflicts in `internal/cloud` then the PR should be good to go

**Comment by malt3** [comment]
> brought config structs back in and marked them as deprecated with a link to new ssh docs. what do you think, @malt3 ?

Looks good! Thanks

**Review by daniel-weisse — APPROVED** [review]

**Pull request merged**
